src/main/java/frc/robot/subsystems/lights/Lights.java

```diff
@@ -0,0 +1,78 @@
+package frc.robot.subsystems.lights;
+
+import com.ctre.phoenix.led.CANdle;
+import com.ctre.phoenix.led.CANdle.LEDStripType;
+import com.ctre.phoenix.led.CANdleConfiguration;
+import com.ctre.phoenix.led.RainbowAnimation;
+import com.ctre.phoenix.led.StrobeAnimation;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.RobotMap;
+import java.util.TreeMap;
+
+public class Lights {
+  private TreeMap<LightCode, Integer[]> lightOptionsMap;
+  private CANdle candle = new CANdle(RobotMap.CANDLE_CAN_ID);
+  private CANdleConfiguration config = new CANdleConfiguration();
+  private LightCode currentLightStatus = LightCode.OFF;
+
+  public enum LightCode{
+    NOTELESS, //red
+    INTAKING, //blue
+    READY_TO_SHOOT, //green
+    NO_TAG, //purple
+    OFF, //grey
+    ALIGNING_TO_TAG, //orange
+    PARTY_MODE;
+  }
+
+  public Lights() {
+    config.stripType = LEDStripType.RGB;
+    config.brightnessScalar = 1.0;
+    candle.configAllSettings(config);
+
+    //array of integers that represents the R, G, B values of each lightcode enum
+    lightOptionsMap = new TreeMap<LightCode, Integer[]>();
+    lightOptionsMap.put(LightCode.NOTELESS, new Integer[] {255, 0, 0});
+    lightOptionsMap.put(LightCode.INTAKING, new Integer[] {0, 0, 255});
+    lightOptionsMap.put(LightCode.ALIGNING_TO_TAG, new Integer[] {0, 255, 0});
+    lightOptionsMap.put(LightCode.NO_TAG, new Integer[] {255, 0, 255});
+    lightOptionsMap.put(LightCode.OFF, new Integer[] {120, 120, 120});
+    lightOptionsMap.put(LightCode.ALIGNING_TO_TAG, new Integer[] {255, 165, 0});
+  }
+
+  public void toggleCode(LightCode light) {
+    if (currentLightStatus == light) {
+      currentLightStatus = LightCode.OFF;
+    } else {
+      currentLightStatus = light;
+    }
+    setLEDs();
+  }
+
+  private void setLEDs() {
+    if (currentLightStatus == LightCode.PARTY_MODE) {
+      setPartyMode();
+    } else if (currentLightStatus == LightCode.NO_TAG) {
+      setNoTag();
```

Comment:
I know you were looking at the minibot Lights class for inspiration. The reason why we needed this second if-statement in the minibot was because NO_TAG was a blinking strobe animation (like how PARTY_MODE is an animation).

if NO_TAG is just a solid color like the other LightCode options (which is what you have commented and in the tree map) then you don't need a separate if statement and function to change the lights to the NO_TAG colorway